hkmc2/shared/src/main/scala/hkmc2/semantics/Symbol.scala

```diff
@@ -65,19 +68,29 @@ class FlowSymbol(label: Str)(using State) extends Symbol:
   override def toString: Str =
     label + State.dbgUid(uid)
 
+  override def subst(using s: SymbolSubst): FlowSymbol = s.mapFlowSym(this)
+
 
-sealed trait LocalSymbol extends Symbol
+sealed trait LocalSymbol extends Symbol:
+  override def subst(using s: SymbolSubst): LocalSymbol
 sealed trait NamedSymbol extends Symbol:
   def name: Str
   def id: Ident
+  override def subst(using s: SymbolSubst): NamedSymbol
 
 abstract class BlockLocalSymbol(name: Str)(using State) extends FlowSymbol(name) with LocalSymbol:
   var decl: Opt[Declaration] = N
+  // HACK: for some reason, if there is no concrete implementation, Scala overrides this subst with the one in
+  // FlowSymbol and then gives a type error because FlowSymbol is not a subtype of BlockLocalSymbol...
+  // Any way to force it to be abstract?
+  // For now, you should be careful when extending BlockLocalSymbol.
+  override def subst(using SymbolSubst): BlockLocalSymbol = lastWords("tried to subst BlockLocalSymbol")
```

Comment:
Yeah, Scala has some annoying limitation like this. This one is due to the requirement that `extends`/`with` clauses need to be checked for validity locally (contrary to self-type annotations). I fixed this by turning the `with LocalSymbol` clause into a self-type annotation.